content/go/server.go.md

```diff
@@ -0,0 +1,522 @@
+	service := services.GetProductPackageService(sess)
+	filter := filter.Build(
+		filter.Path("keyName").
+			Eq(keyname),
+	)
+	mask := "id,keyName,description,items"
+	// Call the method SoftLayer_Product_Package:getAllObjects
+	packages, err := service.Filter(filter).Mask(mask).GetAllObjects()
+	if err != nil {
+		fmt.Printf("%s\n", err)
+		return
+	}
+
+	printItemsDetails(packages[0].Items)
+
+}
+
+/**
+Prints a list of product items.
+*/
+func printItemsDetails(items []datatypes.Product_Item) {
+	rows := []string{"id", "keyname", "description"}
+	cmd := terminal.NewStdUI()
+	table := cmd.Table(rows)
+	for _, item := range items {
+		id := fmt.Sprintf("%d", *item.Id)
+		keyname := sl.Get(item.KeyName).(string)
+		name := sl.Get(item.Description).(string)
+
+		table.Add(id, keyname, name)
+	}
+	cmd.Say("List of product items")
+	table.Print()
+}
+
+/**
+Prints the servers in the account.
+*/
+func listServers() {
+
+	// Get SoftLayer_Account service
+	service := services.GetAccountService(sess)
+
+	// Object-Mask in order to get specific data of server.
+	mask := "id;hostname;primaryIpAddress;primaryBackendIpAddress;datacenter;" +
+		"datacenterName;serviceProvider;hardwareFunctionDescription"
+
+	// Call the getHardware() method.
+	hardware, err := service.Mask(mask).GetHardware()
+	if err != nil {
+		fmt.Printf("\n Unable to get Bare Metal Servers:\n - %s\n", err)
+		return
+	}
+	printServers(hardware)
+
+}
+
+/**
+Prints a list of servers
+*/
+func printServers(servers []datatypes.Hardware) {
+	rows := []string{"id", "hostname", "primary_ip", "backend_ip", "datacenter"}
+	cmd := terminal.NewStdUI()
+	table := cmd.Table(rows)
+	for _, server := range servers {
+		serverId := fmt.Sprintf("%d", server.Id)
+		hostname := sl.Get(server.Hostname).(string)
+		primaryIp := sl.Get(server.PrimaryIpAddress).(string)
+		backendIp := sl.Get(server.PrimaryBackendIpAddress).(string)
+		datacenter := sl.Get(server.DatacenterName).(string)
+
+		table.Add(serverId, hostname, primaryIp, backendIp, datacenter)
+	}
+	cmd.Say("List of Servers")
+	table.Print()
+}
+
+/**
+Prints the available Bare Metal Servers packages.
+*/
+func listServerPackages() {
+
+	// Get SoftLayer_Product_Package_Server service
+	service := services.GetProductPackageServerService(sess)
+
+	// To get all packages of servers available for ordering, we will use the following filter.
+	filter := filter.Build(
+		filter.Path("packageType").
+			In("BARE_METAL_CORE", "BARE_METAL_CPU", "BARE_METAL_CPU_FAST_PROVISION"),
+	)
+	mask := "package"
+	// Call the method SoftLayer_Product_Package_Server::getAllObjects
+	productPackageServes, err := service.Filter(filter).Mask(mask).GetAllObjects()
+	//productPackageServes, err := service.GetAllObjects()
+	if err != nil {
+		fmt.Printf("%s\n", err)
+		return
+	}
+	printPackageDetails(productPackageServes)
+
+}
+
+/**
+Prints the available Bare Metal Servers packages with billing hourly.
+*/
+func listServerPackagesHourly() {
```

Comment:
I am not sure if a bare metal fast provisioning is only for hourly billing, because I see that this package contains item prices hourly too. Please let me see.

Reply:
this function will replace the https://sldn.softlayer.com/go/list_servers_to_order_hourly.go/, but you are right, the package has monthly prices too, so I updated the function name.